Interpreter is currently broken on FreeBSD

Attempting to run this snippet in the interpreter currently fails on FreeBSD:
```
$ crystal i
icr:1:0> require "json"
=> nil
icr:2:0> JSON.parse %({"a": 1})
undefined reference to `iconv_open' (Crystal::Loader::LoadError)
```

Interpreter specs are also failing with similar errors:

```
$ LLVM_CONFIG=/usr/local/bin/llvm-config14 bin/crystal spec --error-trace spec/compiler/interpreter/integration_spec.cr
Failures:

  1) Crystal::Repl::Interpreter integration does CSV
     Failure/Error: interpret(<<-CODE, prelude: "prelude").should eq((1..6).sum.to_s)

       Unhandled exception: undefined reference to `iconv_open' (Crystal::Loader::LoadError)
         from src/compiler/crystal/loader.cr:79:27 in 'find_symbol'
         from src/compiler/crystal/interpreter/context.cr:394:5 in 'c_function'
         from src/compiler/crystal/interpreter/compiler.cr:1969:17 in 'compile_lib_call'
...
  2) Crystal::Repl::Interpreter integration does JSON
     Failure/Error: interpret(<<-CODE, prelude: "prelude").should eq("6")

       Unhandled exception: undefined reference to `iconv_open' (Crystal::Loader::LoadError)
...
  3) Crystal::Repl::Interpreter integration does JSON::Serializable
     Failure/Error: interpret(<<-CODE, prelude: "prelude").should eq("3")

       Unhandled exception: undefined reference to `iconv_open' (Crystal::Loader::LoadError)
         from src/compiler/crystal/loader.cr:79:27 in 'find_symbol'
...
  4) Crystal::Repl::Interpreter integration does IO.pipe (checks that StaticArray is passed correctly to C calls)
     Failure/Error: interpret(<<-CODE, prelude: "prelude").should eq(%("hello"))

       Unhandled exception: undefined reference to `iconv_open' (Crystal::Loader::LoadError)
         from src/compiler/crystal/loader.cr:79:27 in 'find_symbol'
```

The problem is that in FreeBSD `iconv_XXXX` are compatibility symbols, not visible via dlsym(3). Haskell project had the same issue and https://gitlab.haskell.org/ghc/ghc/-/issues/20354 provides extensive background and analysis, I believe similar fix can be applied to Crystal.

--- a/src/lib_c/x86_64-freebsd/c/iconv.cr
+++ b/src/lib_c/x86_64-freebsd/c/iconv.cr
@@ -3,10 +3,13 @@ require "./stddef"
 lib LibC
   type IconvT = Void*
 
-  fun iconv(x0 : IconvT, x1 : Char**, x2 : SizeT*, x3 : Char**, x4 : SizeT*) : SizeT
-  fun iconv_close(x0 : IconvT) : Int
-  fun iconv_open(x0 : Char*, x1 : Char*) : IconvT
+  # Although FreeBSD libc provides iconv_XXXX functions, they are just compatibility symbols,
+  # not directly visible via dlsym(3). Use FreeBSD-specific exported symbols so iconv also
+  # works in the interpreter.
+  fun iconv = __bsd_iconv(x0 : IconvT, x1 : Char**, x2 : SizeT*, x3 : Char**, x4 : SizeT*) : SizeT
+  fun iconv_close = __bsd_iconv_close(x0 : IconvT) : Int
+  fun iconv_open = __bsd_iconv_open(x0 : Char*, x1 : Char*) : IconvT
 
   ICONV_F_HIDE_INVALID = 0x0001
-  fun __iconv(x0 : IconvT, x1 : Char**, x2 : SizeT*, x3 : Char**, x4 : SizeT*, flags : UInt32, invalids : SizeT*) : SizeT
+  fun __iconv = __bsd___iconv(x0 : IconvT, x1 : Char**, x2 : SizeT*, x3 : Char**, x4 : SizeT*, flags : UInt32, invalids : SizeT*) : SizeT
 end

